ikasaneip/topology/src/test/java/org/ikasan/topology/metadata/JsonFlowMetaDataProviderTest.java

```diff
@@ -0,0 +1,236 @@
+package org.ikasan.topology.metadata;
+
+import org.apache.commons.io.IOUtils;
+import org.ikasan.spec.flow.Flow;
+import org.ikasan.spec.flow.FlowConfiguration;
+import org.ikasan.spec.flow.FlowElement;
+import org.ikasan.spec.metadata.FlowMetaData;
+import org.ikasan.topology.metadata.components.*;
+import org.ikasan.topology.metadata.flow.TestFlow;
+import org.ikasan.topology.metadata.flow.TestFlowConfiguration;
+import org.ikasan.topology.metadata.flow.TestFlowElement;
+import org.junit.Assert;
+import org.junit.Test;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.HashMap;
+import java.util.Map;
+
+
+public class JsonFlowMetaDataProviderTest
+{
+    public static final String SIMPLE_FLOW_RESULT_JSON = "/data/simpleFlow.json";
+    public static final String MULTI_RECIPIENT_FLOW_RESULT_JSON = "/data/multiRecipientFlow.json";
+    public static final String SINGLE_RECIPIENT_FLOW_RESULT_JSON = "/data/singleRecipientFlow.json";
+
+    @Test
+    public void test_simple_flow() throws IOException
+    {
+        JsonFlowMetaDataProvider jsonFlowMetaDataProvider = new JsonFlowMetaDataProvider();
+        String json = jsonFlowMetaDataProvider.describeFlow(createSimpleFlow());
+
+        Assert.assertEquals("JSON Result must equal!", loadDataFile(SIMPLE_FLOW_RESULT_JSON).replaceAll("\r", "").replaceAll("\n", "")
```

Comment:
two options in here,
1) use lib which will help with compering and asserting JSON,  like https://www.baeldung.com/jsonassert
2) dont use a pretty print when you serialising object to JSON string., or allow injection of boolean flag which you can utilise in a test

Reply:
@andrzej-majewski Take a look now.